packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx

```diff
@@ -105,7 +105,10 @@ class AudioPlayerWithYoutubeSpotify extends Component {
     const newSourceAlbumInfo = albumSpotifyYoutubeInfo[newSource];
     const noAlbumWithNewSource = currentTrack === 0 && !newSourceAlbumInfo;
     const noTrackWithNewSource = !tracklistToShow.find(f => currentTrack === f.trackNumber);
-    const trackSelected = (noAlbumWithNewSource || noTrackWithNewSource) ? 1 : currentTrack;
+
+    const firstTrackAvailable = tracklistToShow.find(f => f.trackNumber === 1) ? tracklistToShow.find(f => f.trackNumber === 1) : head(tracklistToShow);
```

Comment:
SUGGESTION: Just store the `.find` results so you don't have to duplicate the code.

Reply:
thanks for finding